files/default/neutron-ha-tool.py

```diff
@@ -160,15 +167,38 @@ def run(args):
                   "or from %s; aborting!" % OS_PASSWORD_FILE)
         sys.exit(1)
 
-    # instantiate client
-    qclient = client.Client('2.0', auth_url=os.environ['OS_AUTH_URL'],
-                            username=os.environ['OS_USERNAME'],
-                            tenant_name=os.environ['OS_TENANT_NAME'],
-                            password=os_password,
-                            endpoint_type='internalURL',
-                            region_name=os.environ['OS_REGION_NAME'],
-                            insecure=args.insecure,
-                            ca_cert=ca)
+    auth_version = os.getenv('OS_AUTH_VERSION', None)
+    if not auth_version:
+        auth_version = os.getenv('OS_IDENTITY_API_VERSION', None)
+        if not auth_version:
+            auth_version = 3
+
+    kclient = IDENTITY_API_VERSIONS[auth_version]
+    kclient_kwargs = dict()
+    kclient_kwargs['username'] = os.environ['OS_USERNAME']
+    kclient_kwargs['password'] = os.environ['OS_PASSWORD']
+    kclient_kwargs['insecure'] = args.insecure
+    kclient_kwargs['ca_cert'] = ca
+    kclient_kwargs['auth_url'] = os.environ['OS_AUTH_URL']
+
+    tenant_name = os.getenv('OS_TENANT_NAME')
+    if tenant_name and auth_version != 3:
+        kclient_kwargs['tenant_name'] = tenant_name
+    else:
+        kclient_kwargs['project_name'] = os.environ['OS_PROJECT_NAME']
+
+    # Instantiate Keystone client
+    keystone = kclient.Client(**kclient_kwargs)
+
+    # Instantiate Neutron client
+    qclient = nclient.Client(
+        '2.0',
+        endpoint_url=keystone.service_catalog.url_for(
+            service_type='network',
+            endpoint_type=os.environ['OS_ENDPOINT_TYPE']
```

Comment:
I'm nearly sure that our crowbar cookbook doesn't set this environment variable in the pacemaker resource (old code was using `internalURL` hard-coded, it seems).